mantidproject/mantid
# Add support for sip-build to CMake

**Description of work.**

Updates our cmake support for both versions of the sip build system. This will allow us to build with the HEAD of brew but also support the other operating systems.

**To test:**

* Run `brew upgrade`
  * This will bring down Python 3.9. Check this is what you see when the install finishes. You might need to restart the shell if not.
  * Run `python3 -m pip install -U -r /usr/local/Homebrew/Library/Taps/mantidproject/homebrew-mantid/requirements.txt` to reinstall the Python requirements.
* Try a clean build of mantid.

Fixes #30912 

*This does not require release notes* because **it is a developer change.**



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
